### PR TITLE
Project map

### PR DESCRIPTION
**Pull request by bazuzi**
- add a page at /hq/admin/project_map/ that displays an interactive map of current projects and provides information by country and by individual project
- allow filtering by project facets, identical to /hq/admin/domains/

google doc with planning info [here](https://docs.google.com/a/dimagi.com/document/d/1JF1dhN0vG0AQsLJ5Csnp1OLQ8kaUlPld1B_46ASVFF0/edit?usp=sharing):

@czue @amstone326 @emord

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 11 errors:

* corehq/apps/hqadmin/reports.py, line 826 - E302 expected 2 blank lines, found 1
* corehq/apps/hqadmin/reports.py, line 827 - E225 missing whitespace around operator
* corehq/apps/hqadmin/reports.py, line 845 - E501 line too long (119 > 115 characters)
* corehq/apps/hqadmin/reports.py, line 853 - W601 .has_key() is deprecated, use 'in'
* corehq/apps/hqadmin/reports.py, line 853 - E501 line too long (118 > 115 characters)
* corehq/apps/hqadmin/reports.py, line 856 - F841 local variable 'first_form_default_message' is assigned to but never used
* corehq/apps/hqadmin/static/hqadmin/js/project_map.js, line 58 - 'spread/rest operator' is only available in ES6 (use esnext option).
* corehq/apps/hqadmin/static/hqadmin/js/project_map.js, line 147 - Missing semicolon.
* corehq/apps/hqadmin/static/hqadmin/js/project_map.js, line 171 - A leading decimal point can be confused with a dot: '.9'.
* corehq/apps/hqadmin/static/hqadmin/js/project_map.js, line 276 - Missing semicolon.
* corehq/apps/hqadmin/static/hqadmin/js/project_map.js, line 279 - Missing semicolon.

File: corehq/apps/hqadmin/reports.py [hunk 1]
```diff
@@ -824,6 +824,54 @@ def format_bool(val):
...
+        super(AdminDomainStatsReport, self).set_bootstrap3_status(request, *args, **kwargs)
+
+    # a modified version of AdminDomainStatsReport.rows
+    @property
+    def rows(self):
+        domains = [res['_source'] for res in self.es_results.get('hits', {}).get('hits', [])]
+
+        def format_date(dstr, default):
+            # use [:19] so that only only the 'YYYY-MM-DDTHH:MM:SS' part of the string is parsed
+            return datetime.strptime(dstr[:19], '%Y-%m-%dT%H:%M:%S').strftime('%Y/%m/%d %H:%M:%S') if dstr else default
```
Comment (dimagimon) [inline corehq/apps/hqadmin/reports.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/hqadmin/reports.py [hunk 2]
```diff
@@ -824,6 +824,54 @@ def format_bool(val):
...
+            if isinstance(val, bool):
+                return u"{}".format(val)
+            return _('No info')
+
+        for dom in domains:
+            # for some reason when using the statistical facet, ES adds an empty dict to hits
+            if 'name' in dom:
+                first_form_default_message = _("No Forms")
+                if dom.get("cp_last_form", None):
+                    first_form_default_message = _("Unable to parse date")
```
Comment (dimagimon) [inline corehq/apps/hqadmin/reports.py]: F841 local variable 'first_form_default_message' is assigned to but never used

File: corehq/apps/hqadmin/reports.py [hunk 3]
```diff
@@ -824,6 +824,46 @@ def format_bool(val):
...
+
+    # a modified version of AdminDomainStatsReport.rows
+    @property
+    def rows(self):
+        domains = [res['_source'] for res in self.es_results.get('hits', {}).get('hits', [])]
+
+        def format_date(dstr, default):
+            # use [:19] so that only only the 'YYYY-MM-DDTHH:MM:SS' part of the string is parsed
+            return datetime.strptime(dstr[:19], '%Y-%m-%dT%H:%M:%S').strftime('%Y/%m/%d %H:%M:%S') \
+                   if dstr else default
```
Comment (dimagimon) [inline corehq/apps/hqadmin/reports.py]: E126 continuation line over-indented for hanging indent

**Comment by emord** [comment]
Looks good once tests pass. Doesn't look too difficult to anonymize so that we could show publicly as the doc describes.

**Comment by czue** [comment]
can't wait to check this out! thanks for the awesome work @bazuzi ! we'll miss you  :)

**Pull request merged**